validator/src/main/java/com/amazon/aoc/helpers/CWMetricHelper.java

```diff
@@ -20,57 +20,130 @@
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.*;
 
-/**
- * helper class for getting expected metrics from templates.
- */
+/** helper class for getting expected metrics from templates. */
 public class CWMetricHelper {
   private static final String DEFAULT_DIMENSION_NAME = "OTelLib";
   MustacheHelper mustacheHelper = new MustacheHelper();
 
   /**
    * get expected metrics from template with injecting context.
    *
-   * @param context        testing context
+   * @param context testing context
    * @param expectedMetric expected template
-   * @param caller         http caller, none caller, could be null
+   * @param caller http caller, none caller, could be null
    * @return list of metrics
    * @throws Exception when caller throws exception or template can not be found
    */
   public List<Metric> listExpectedMetrics(
-      Context context,
-      FileConfig expectedMetric,
-      ICaller caller
-  ) throws Exception {
+      Context context, FileConfig expectedMetric, ICaller caller) throws Exception {
     // call endpoint
     if (caller != null) {
       caller.callSampleApp();
     }
 
     // get expected metrics as yaml from config
-    String yamlExpectedMetrics = mustacheHelper.render(expectedMetric, context);
 
-    // load metrics from yaml
+    String yamlExpectedMetrics = null;
+    List<Metric> expectedMetricList = null;
     ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-    List<Metric> expectedMetricList = mapper.readValue(
-          yamlExpectedMetrics.getBytes(StandardCharsets.UTF_8),
-          new TypeReference<List<Metric>>() {});
+
+    if (context != null
+        && context.getTestcase() != null
+        && context.getTestcase().contains("hostmetrics_receiver")) {
+      if (context.getHostmetricsContext() != null) {
+        yamlExpectedMetrics = mustacheHelper.render(expectedMetric, context);
+        List<com.amazon.aoc.models.cwmodel.Metric> expectedMetricsForHostMetrics =
+            mapper.readValue(
+                yamlExpectedMetrics.getBytes(StandardCharsets.UTF_8),
+                new TypeReference<List<com.amazon.aoc.models.cwmodel.Metric>>() {});
+        expectedMetricList = this.parseCWModel(mapper, expectedMetricsForHostMetrics);
+      }
+    } else {
+      yamlExpectedMetrics = mustacheHelper.render(expectedMetric, context);
+      expectedMetricList =
+          mapper.readValue(
+              yamlExpectedMetrics.getBytes(StandardCharsets.UTF_8),
+              new TypeReference<List<Metric>>() {});
+    }
 
     if (context.getIsRollup()) {
       return this.rollupMetric(expectedMetricList);
     }
-
     return expectedMetricList;
   }
 
+  private List<Metric> parseCWModel(
```

Comment:
```suggestion
  private List<Metric> convertHostMetricModeltoMetricModel(
```
I believe a more descriptive name is helpful here.